internal/packing/packing.go

```diff
@@ -178,6 +152,8 @@ func alloc(n *Node, width, height int) *Node {
 				parent: n,
 			}
 		}
+		// Note: it now MUST fit, as child0 is bigger than width * height when getting here.
+		// Thus, alloc can't return nil, but it may do another split along the other dimension.
```

Comment:
I don't understand this comment. What does width * height mean? Can we replace this with an if condition with a panic as an assertion?

Reply:
It means width x height. But yes, I can add a panic here to document this better.

The reason why this is important is that if this alloc call fails, we'll have mutated the tree WITHOUT actually allocating anything, which could be a mess during the extend process.

Reply:
OK, let's add assertions here.

We can also add an assertion to check that alloc returns a non-nil value here.

Reply:
Added a panic and more explanation.